### ISSUE TITLE
Redl current does not work with SplineProfile current

### ISSUE THREAD

**Issue by daringli** [PlasmaControl/DESC#1650]
Rewriting the final function in _bootstrap.py as seen below should work, but it doesn't as it gives nan in the objective when solving for the equilibrium. In addition, if the spline has a knot at zero, it will result in a nan which will cause the cumulative_trapezoid to return all nans.
```python
if isinstance(profiles["current"],PowerSeriesProfile):
        degree = kwargs.get(
            "degree",
            min(
                (
                    profiles["current"].basis.L
                    if profiles["current"] is not None
                    else transforms["grid"].num_rho - 1
                ),
                transforms["grid"].num_rho - 1,
            ),
        )
        XX = jnp.vander(rho, degree + 1)[:, :-1]  # remove constant term
        c_l_r = jnp.pad(jnp.linalg.lstsq(XX, current_r)[0], (0, 1))  # manual polyfit
        c_l = jnp.polyint(c_l_r)
        current = jnp.polyval(c_l, rho)

    elif isinstance(profiles["current"], SplineProfile):
        current = cumulative_trapezoid(y=current_r, x=rho, initial=0.0)
```

**Comment by dpanici** [comment]
For reference, this issue specifically was because the `current Redl` compute variable had a bug where its if statement for checking if a profile were a `PowerSeriesProfile` actually would fail for non power series profiles, `SplineProfile` included

The NaN issue being mentioned here was due to the fact that the `current Redl` compute function is not defined on-axis, and so when `rho=0` is included, it yields a `nan` in `<J*B> Redl` which pollutes the entire output with `nan` since we then fit that quantity with a power series in order to integrate `current_r Redl` and return `current Redl`. The scripts being used by the user were computing the Redl formula at the axis.

Note also that the formula is invalid for points where the kinetic profiles vanish (due to divisions in the formula), and so if a profile goes to zero at `rho=1`, there would be a nan at `rho=1` as well when computing `current Redl`
